File: go/alloc_heavy.go
package main
import (
	"fmt"
	"strconv"
	"strings"
	"time"
)

func main() {
	start := time.Now()
	var items []string
	for i := 0; i < 750000; i++ {
		items = append(items, strings.Repeat("item-", 2) + strconv.Itoa(i))
	}
	fmt.Println("alloc_heavy", len(items))
	fmt.Printf("Time: %v\n", time.Since(start))
}

File: python/alloc_heavy.py
import gc
import time
import gcstats  # noqa: F401  (registers stats at exit)

start = time.perf_counter()
items = []
for i in range(750000):
    items.append("item-" * 2 + str(i))
elapsed = time.perf_counter() - start
print("alloc_heavy", len(items))
print("Time:", f"{elapsed:.3f}", "seconds")